### PR TITLE
Examples for Tutorial 3 – Observing the Player

### PR DESCRIPTION
**Pull request by BohdanBilonoh**
Need to add examples for Tutorial 3 – Observing the Player from [doc](https://docs.google.com/document/d/19cVTBPadBSYWdEf6f933mdOQCrPgMU_VdG1EaTNyb2c/edit)

[JIRA Issue](https://jira.ops.aol.com/browse/OMSDK-539)

### CODE REVIEW

File: NewTutorials/Tutorial/PlayerViewControllerWrapper.swift [hunk 1]
```diff
@@ -77,6 +79,17 @@ class PlayerViewControllerWrapper: UIViewController {
                     return props
                 }
                 
+                if props.shouldObserve {
+                    _ = player.addObserver(on: DispatchQueue.global(qos: .background)) { (props) in
+                        guard let item = props.playbackItem else { return }
+                        let isStreamPlaying = item.ad.isStreamPlaying || item.content.isStreamPlaying
+                        let currentTime = item.ad.time.static?.current ?? item.content.time.static?.current ?? 0
+                        
+                        print("isPlaying:", isStreamPlaying)
```
Comment (AndriiMoskvin) [inline NewTutorials/Tutorial/PlayerViewControllerWrapper.swift]: Can you render these changes on the UI as we discussed yesterday?

**Review by AndriiMoskvin — CHANGES_REQUESTED** [review]

File: NewTutorials/Tutorial/PlayerViewControllerWrapper.swift (hunk 1 above)
Comment (BohdanBilonoh) [reply NewTutorials/Tutorial/PlayerViewControllerWrapper.swift]: Of course. Planned to add it today

**Comment by AndriiMoskvin** [comment]
@BogdanBilonog Ping me when this will be finished.

**Comment by BohdanBilonoh** [comment]
@AndriiMoskvin Can be reviewed

File: NewTutorials/Tutorial/PlayerViewControllerWrapper.swift [hunk 2]
```diff
@@ -11,108 +11,158 @@ class PlayerViewControllerWrapper: UIViewController {
...
+            contentPlayer.item = .playable(controls)
+            
+            var props = props
+            props = .player(contentPlayer)
+            
+            return props
+        }
+        
+        playerPropsObserverDispose = player.addObserver { [weak self, weak player] in
+            self?.props.stats = map(props: $0)
```
Comment (AndriiMoskvin) [inline NewTutorials/Tutorial/PlayerViewControllerWrapper.swift]: 💣

File: NewTutorials/Tutorial/PlayerViewControllerWrapper.swift [hunk 3]
```diff
@@ -11,108 +11,158 @@ class PlayerViewControllerWrapper: UIViewController {
...
+            if map(props: $0) { player?.selectVideo(atIndex: 0) }
```
Comment (AndriiMoskvin) [inline NewTutorials/Tutorial/PlayerViewControllerWrapper.swift]: 💣 🔥

**Review by AndriiMoskvin — CHANGES_REQUESTED** [review]

File: NewTutorials/Tutorial/PlayerViewControllerWrapper.swift [hunk 4]
```diff
@@ -11,108 +11,158 @@ class PlayerViewControllerWrapper: UIViewController {
...
+            contentPlayer.playlist?.next = customNextCommand
+            contentPlayer.item = .playable(controls)
+            
+            var props = props
+            props = .player(contentPlayer)
+            
+            return props
+        }
+        
+        playerPropsObserverDispose = player.addObserver { [weak self, weak player] in
```
Comment (AndriiMoskvin) [inline NewTutorials/Tutorial/PlayerViewControllerWrapper.swift]: Are you sure that **ALL** this weak references are required?

**Review by AndriiMoskvin — CHANGES_REQUESTED** [review]

**Review by AndriiMoskvin — DISMISSED** [review]

**Review by AndriiMoskvin — APPROVED** [review]

**Pull request merged**